libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -936,29 +606,33 @@ void ArrayUtils::clearStorageLoop(Type const* _type) const
 				_context << Instruction::POP;
 				return;
 			}
-			// stack: end_pos pos
-
+			// stack: start_pos slotCount
+			// Initialize loop counter i = 0
+			_context << u256(0);
+			// stack: start_pos slotCount i
```

Comment:
You could reuse `slotCount` as a counter if you switched to counting down. This would also simplify the code a bit (e.g. the condition would reduce to `ISZERO`).